src/main/java/com/google/api/codegen/discovery/config/js/JSTypeNameGenerator.java

```diff
@@ -35,4 +35,10 @@ public String stringDelimiter() {
   public String getStringFormatExample(String format) {
     return getStringFormatExample(format, "Date.toISOString()", "Date.toISOString()");
   }
+
+  @Override
+  public String getDiscoveryDocUrl(String apiName, String apiVersion) {
+    return String.format(
+        "https://content.googleapis.com/discovery/v1/apis/%s/%s/rest", apiName, apiVersion);
```

Comment:
The other client libraries are not dynamic, so don't include the discovery doc URL implicitly. Python constructs the URL on the fly from the API name and version, so we don't include it there for any Google API. For JavaScript though the recommended method of constructing the service is to actually pass in the discovery URL, so we have to include it.

It was already there in the snippet, but I moved it to the `SampleConfig`/`JSTypeNameGenerator` so we can target a custom discovery doc in the dynamic clients.

So no other language uses this by default, which is why only JS has it, and Python only considers it if it has been explicitly overridden. That is, by default for any discovery doc the string will be empty, and the python build statement will look like:

```discovery.build('foo', 'v1')```

and if it's specified, it'll look like:

```discovery.build('foo', 'v1', discoveryServiceUrl='{discoveryDocUrl}'```

None of the other languages use this property.